### ISSUE TITLE
refactor(deps): `ent` dependency

### ISSUE THREAD

**Issue by danielbankhead** [googleapis/nodejs-storage#1907]
Remove `ent` & `@types/ent` dependencies - they do not seem necessary given the context of its usage. 

Perhaps [`decodeURI`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURI) would work as an alternative.

**Comment by ddelgrosso1** [comment]
I do not believe this to be easy to remove. The reason it is imported is that under certain situations (rate limiting if I remember correctly) the GCS API will return a response in HTML. Currently Ent is used to decode the returned HTML. I spoke with other client library owners and while they do not parse the HTML like Node does, if we were to remove this it would be a breaking change. We have two choices, close this issue as will not fix or move it until our next major release which is tentatively scheduled for Spring 2023.

**Comment by luc122c** [comment]
Hi, I think this needs revisting. [ent](https://www.npmjs.com/package/ent?activeTab=code) relies on the [deprecated punycode module](https://nodejs.org/api/deprecations.html#DEP0040). This is now causing issues in Node v21.x.

**Comment by ddelgrosso1** [comment]
Hi @luc122c thanks for the heads up. I will take another look and see if there is a way we can remove `ent` without causing a breaking change.

**Comment by drichardson** [comment]
Not sure if it's a drop in replacement, but perhaps https://www.npmjs.com/package/entities could be used to replace ent (which hasn't been updated in 9 years).

**Comment by ddelgrosso1** [comment]
@drichardson thanks for the heads up. I will poke around in entities to see if it meets our needs.

**Comment by drichardson** [comment]
@ddelgrosso1 Whoops I just saw your note after making a branch using a different library called [he](https://www.npmjs.com/package/he) which is pretty old and boring which seemed good for this 😄.

**Comment by cfdavidpetter** [comment]
@luc122c  @ddelgrosso1 
I'm having trouble with this in today's latest version.

`FROM node:latest`

**Comment by cfdavidpetter** [comment]
@ddelgrosso1 my problem is this:

`(node:8) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
    at node:punycode:3:9
    at BuiltinModule.compileForInternalLoader (node:internal/bootstrap/realm:398:7)
    at BuiltinModule.compileForPublicLoader (node:internal/bootstrap/realm:337:10)
    at loadBuiltinModule (node:internal/modules/helpers:104:7)
    at Module._load (node:internal/modules/cjs/loader:999:17)
    at Module.require (node:internal/modules/cjs/loader:1230:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (/app/node_modules/whatwg-url/lib/url-state-machine.js:2:18)
    at Module._compile (node:internal/modules/cjs/loader:1368:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1426:10)`

**Comment by ddelgrosso1** [comment]
Thanks @cfdavidpetter we are aware of the deprecation of `punycode` and plan to remove `ent` in the next major version of the storage library.

**Comment by danielbankhead** [comment]
Related:
- https://github.com/googleapis/nodejs-common/pull/813

**Comment by ddelgrosso1** [comment]
https://github.com/googleapis/nodejs-storage/pull/2451
